### PR TITLE
Unused values as warnings

### PR DESCRIPTION
**Pull request by bender2k14**
First commit modifies `fsproj` files to turn unused values into build warnings.  The second commit deletes one unused line of test code.  The third commit renames all unused values to `_`.

### CODE REVIEW

**Comment by bender2k14** [comment]
> ...I have often found it equally (if not more) nice to know at a glance what a parameter actually is, even if unused, and names help with this. For example, it becomes easier to make use of a parameter later when needed.

I think code should be optimized for reading since that is the typical case.  If someone is going to write new code in that area, then they can look at the type to quickly determine "what a parameter actually is".

In this case of `CmdParam`, the type of the parameter is just `obj`.  For lambda function, the convention I see (both in this project and beyond) is to have the identify be the first letter of the corresponding type.  So for this case, I agree that replacing `p` with `_` makes things significantly less clear when _not_ reading.  However, I think the fix is to give a proper type to the parameter.  See #146.

**Comment by bender2k14** [comment]
> In any case, given that I don't think it's generally a given which option to prefer, I have never turned on `--warnon:1182`. (This is also combined with the fact that I usually always turn on `<WarningsAsErrors />`, forcing me to fix warnings.)

I agree that `<WarningsAsErrors />` is a good idea, but sadly it is currently broken :(
https://github.com/dotnet/core-sdk/issues/1708

**Comment by bender2k14** [comment]
> For example, in the samples, I think I generally prefer names...

A simple compromise would be to only add `--warnon:1182` in non-sample projects.

File: src/Samples/UiBoundCmdParam/Program.fs [hunk 1]
```diff
@@ -26,7 +26,7 @@ let bindings () : Binding<Model, Msg> list = [
   "Numbers" |> Binding.oneWay(fun m -> m.Numbers)
   "Limit" |> Binding.twoWay((fun m -> float m.EnabledMaxLimit), int >> SetLimit)
   "Command" |> Binding.cmdParamIf(
-    (fun p m -> Command),
+    (fun _ _ -> Command),
```
Comment (cmeeren) [inline src/Samples/UiBoundCmdParam/Program.fs]: I'd like this reverted.

My reasoning is that the names of even unused parameters in bindings can make signatures/intent more clear to newcomers (who are the primary target audience for the samples).

File: src/Samples/UiBoundCmdParam/UiBoundCmdParam.fsproj [hunk 2]
```diff
@@ -4,6 +4,8 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp3.0</TargetFramework>
     <UseWpf>true</UseWpf>
+    <!--Turn on warnings for unused values (arguments and let bindings) -->
+    <OtherFlags>$(OtherFlags) --warnon:1182</OtherFlags>
```
Comment (cmeeren) [inline src/Samples/UiBoundCmdParam/UiBoundCmdParam.fsproj]: I'd like all of these reverted in samples

**Comment by cmeeren** [comment]
You make good sense. I can agree that code should be optimized for reading. I can accept the changes to the main project.

I can also accept most of the changes to the samples. Apart from the `--warnon:1182` you mentioned,   there are actually just a single change I'd really like reverted. I've left a comment.

Also, instead of `<OtherFlags>$(OtherFlags) --warnon:1182</OtherFlags>` I *think* there may be a `WarnOn` element of sorts that can be used to add codes more directly (though I can't find it right now, so I may be mistaken). Could you have a quick look? (Don't spend your entire evening on it.)

File: src/Samples/UiBoundCmdParam/UiBoundCmdParam.fsproj (hunk 2 above)
Comment (bender2k14) [reply src/Samples/UiBoundCmdParam/UiBoundCmdParam.fsproj]: Done

File: src/Samples/UiBoundCmdParam/Program.fs (hunk 1 above)
Comment (bender2k14) [reply src/Samples/UiBoundCmdParam/Program.fs]: Sure.  Done.

**Comment by bender2k14** [comment]
I would prefer to squash some commits before merging

**Comment by cmeeren** [comment]
Let me know when you're ready!

**Comment by bender2k14** [comment]
> Also, instead of `<OtherFlags>$(OtherFlags) --warnon:1182</OtherFlags>` I think there may be a `WarnOn` element of sorts that can be used to add codes more directly (though I can't find it right now, so I may be mistaken). Could you have a quick look? (Don't spend your entire evening on it.)

This is the only way that I know how to do it in the project file.

I have found this part of devops rather confusing.  On the one had, there is very clear information about what [options `fsc`, the F# compiler, accepts](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/compiler-options).  On the other hand, us developers call commands like `dotnet build` that have their own default arguments, extract arguments from the project files, and eventually call the F# compiler or one of the C# compilers.  What seems to be missing is documentation specifying all of the translations from `dotnet build` to `fsc`.

**Comment by bender2k14** [comment]
Ready :)

**Comment by cmeeren** [comment]
Thanks a lot!

**Pull request merged**